includes/class-newspack-popups-inserter.php

```diff
@@ -273,32 +273,19 @@ public static function insert_popups_amp_access( $popups ) {
 	}
 
 	/**
-	 * Disable popups on non-AMP pages with a form with POST method.
-	 * More context: https://github.com/ampproject/amphtml/issues/27638.
+	 * Disable popups on posts and pages which have newspack_popups_has_disabled_popups.
 	 *
-	 * @param string $content The content of the post.
 	 * @return bool True if popups should be disabled for current page.
 	 */
-	public static function assess_amp_form_issue( $content = false ) {
-		if ( is_admin() || ! is_singular() ) {
-			return false;
-		}
-		if ( ! $content ) {
-			$content = get_the_content();
-		}
-		$is_amp = function_exists( 'is_amp_endpoint' ) && is_amp_endpoint();
-
-		if ( ! $is_amp ) {
-			return preg_match( "/<form.*method=[\"']post[\"']/", $content );
-		}
-		return false;
+	public static function assess_has_disabled_popups() {
+		return get_post_meta( get_the_ID(), 'newspack_popups_has_disabled_popups', true );
```

Comment:
Could you add a filter here so that plugins and themes can also influence this assessment? It will give us a lot of flexibility to handle similar problems in the future.

Reply:
Done in 687609762cc98e83b6bcc5bad0e3fefd14565528